crates/oxc_linter/src/rules/eslint/no_regex_spaces.rs

```diff
@@ -228,7 +228,7 @@ fn test() {
         "var foo = RegExp(' [  ] [  ] ');",
         r"var foo = new RegExp(' \[   ');",
         r"var foo = new RegExp(' \[   \] ');",
-        r"var foo = /[\\q{    }]/v;",
```

Comment:
It looks like the Regexp it represents is `/[\q{    }]/v`.

```js
console.log("/[\\q{    }]/v")
```

Reply:
> console.log("/[\\q{    }]/v")

This is not evaluated as `RegExp`, this is just a `String`. 👀 

![image](https://github.com/user-attachments/assets/5c77bc65-7238-4cd7-9dbb-01b201ceabfc)

Reply:
Maybe I missed some information, but if I understand correct, the test case below 🤔

```js
{ code: "var foo = /[\\q{    }]/v;", languageOptions: { ecmaVersion: 2024 } },
```

is equal to the file content: 

```js
var foo = /[\q{    }]/v;
```

This is what we write in JS file. I tested it on NodeJS:

```js
// filename: a.js
const fs = require('fs')
const assert = require('assert')
let foo = "var foo = /[\\q{    }]/v;\n"

let input = fs.readFileSync('input.js', 'utf8');
assert.strictEqual(foo, input)

// filename: input.js
var foo = /[\q{    }]/v;
```

I know this because I've fallen into a similar trap: #2085

Reply:
I may also missed something, 😅

But I'd like to fix here is Rust <-> Rust conversion, not Rust <-> JS.

```rs
r"var foo = /[\\q{    }]/v;",
```

By leading `r"..."`, we do not need to escape `\` unless we really need 2 `\`s.

And if there are 2 `\`(= original code), it's not a valid `RegExp` syntax.

Reply:
In rust side, `r"var foo = /[\q{    }]/v;"` is equal to the test case in the original ESLint test case, they all represent the same JS file which content is:

![CleanShot 2024-08-03 at 13 41 41@2x](https://github.com/user-attachments/assets/fab51ac7-083d-4b20-a162-421bf7073207)

Reply:
> if we port it over to our side, it can be `r"var foo = /[\q{    }]/v;"` or `"var foo = /[\\q{    }]/v;"`. 👀

Yes. 

But it was `r"var foo = /[\\q{ }]/v;"`. So I've fixed it by just removing the leading `r` now.

(In my first commit, I tried to fix this by removing the extra `\`. I also added the `NestedClass` pattern to expand the test case. This may have led to misunderstandings. Now I have minimized the diff for sure~.)